### PR TITLE
Add multinetworkpolicy unit tests

### PR DESCRIPTION
**Pull request by sebrandon1**
Added unit tests for `multinetworkpolicy` file.

I made notes in the `GetTestClients()` func that explain that I don't exactly think their fake client actually works the way we think it works.  It does not seem to accept runtimeObjects the way we expect with other libraries.

The `multinetworkpolicy` repo itself is entirely way out of date and not maintained anymore so I'm not sure it's worth exploring further.

### CODE REVIEW

File: pkg/networkpolicy/multinetworkpolicy_test.go [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+	"testing"
+
+	"github.com/k8snetworkplumbingwg/multi-networkpolicy/pkg/apis/k8s.cni.cncf.io/v1beta1"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestNewMultiNetworkPolicyPull(t *testing.T) {
```
Comment (kononovn) [inline pkg/networkpolicy/multinetworkpolicy_test.go:13]: Please rename this test function to TestPullMultiNetworkPolicy

File: pkg/networkpolicy/multinetworkpolicy_test.go [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+
+	result := testBuilder.WithPolicyType(v1beta1.PolicyTypeEgress)
+
+	assert.Equal(t, v1beta1.PolicyTypeEgress, result.Definition.Spec.PolicyTypes[0])
+
+	result = testBuilder.WithPolicyType("")
+
+	assert.Equal(t, "The policy Type is an empty string", result.errorMsg)
+}
+
```
Comment (kononovn) [inline pkg/networkpolicy/multinetworkpolicy_test.go:500]: Please add test function for Exists()

**Review by kononovn — CHANGES_REQUESTED** [review]
Please check few comments above.

File: pkg/networkpolicy/multinetingressrule_test.go [hunk 3]
```diff
@@ -0,0 +1 @@
+package networkpolicy
```
[the remark this answers is not in the document]
Comment (sebrandon1) [reply pkg/networkpolicy/multinetingressrule_test.go:1]: Done.  I'm working on these next so I'll add it there.

File: pkg/networkpolicy/multinetegressrule_test.go [hunk 4]
```diff
@@ -0,0 +1 @@
+package networkpolicy
```
[the remark this answers is not in the document]
Comment (sebrandon1) [reply pkg/networkpolicy/multinetegressrule_test.go:1]: Done.  I'm working on these next so I'll add it there.

File: pkg/networkpolicy/multinetworkpolicy_test.go [hunk 5]
```diff
@@ -0,0 +1,516 @@
...
+		if testCase.expected {
+			assert.Equal(t, testCase.podSelector, testBuilder.Definition.Spec.PodSelector.MatchLabels)
+		} else {
+			assert.Nil(t, testBuilder.Definition.Spec.PodSelector.MatchLabels)
+		}
+	}
+}
+
+func TestMultiNetworkPolicyWithNetwork(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (sebrandon1) [reply pkg/networkpolicy/multinetworkpolicy_test.go:429]: Done.

File: pkg/networkpolicy/multinetworkpolicy_test.go (hunk 1 above)
Comment (sebrandon1) [reply pkg/networkpolicy/multinetworkpolicy_test.go:13]: Done.

File: pkg/networkpolicy/multinetworkpolicy_test.go (hunk 2 above)
Comment (sebrandon1) [reply pkg/networkpolicy/multinetworkpolicy_test.go:504]: Done.

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**